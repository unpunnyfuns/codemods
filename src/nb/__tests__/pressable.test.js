import { describe, expect, it } from 'vitest'
import { testTransform } from '../../__tests__/test-helper.js'

describe('nb/pressable', () => {
  it('migrates Pressable to React Native with StyleSheet and default accessibilityRole', () => {
    const output = testTransform('nb/pressable', 'nb/pressable', 'js')
    expect(output).toMatchSnapshot()
  })
})
