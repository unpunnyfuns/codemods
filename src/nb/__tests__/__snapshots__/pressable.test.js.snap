// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`nb/pressable > migrates Pressable to React Native with StyleSheet and default accessibilityRole 1`] = `
"import { Text, Pressable, StyleSheet } from 'react-native';

import { color } from '@hb-frontend/nordlys';

export function MyComponent() {
  return (
    <>
      <Pressable onPress={() => {}} accessibilityRole='button' style={styles.pressable0}>
        <Text>Simple pressable</Text>
      </Pressable>
      <Pressable accessibilityRole="link" onPress={() => {}} style={styles.pressable1}>
        <Text>Custom role</Text>
      </Pressable>
      <Pressable onPress={() => {}} accessibilityRole='button'>
        <Text>With pseudo props</Text>
      </Pressable>
    </>
  );
}

const styles = StyleSheet.create({
  pressable0: {
    backgroundColor: color.blue['500'],
    padding: 4,
    margin: 2
  },

  pressable1: {
    marginTop: 8
  }
});
"
`;
